ILSpy/Docking/DockWorkspace.cs

```diff
@@ -45,14 +46,46 @@ public class DockWorkspace : INotifyPropertyChanged, ILayoutUpdateStrategy
 
 		public event PropertyChangedEventHandler PropertyChanged;
 
-		public static DockWorkspace Instance { get; } = new DockWorkspace();
+		public static DockWorkspace Instance { get; private set; }
 
-		private DockWorkspace()
+		internal DockWorkspace(MainWindow parent)
 		{
+			Instance = this;
 			this.TabPages.CollectionChanged += Documents_CollectionChanged;
+			parent.CurrentAssemblyListChanged += MainWindow_Instance_CurrentAssemblyListChanged;
 		}
 
-		private void Documents_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+		private void MainWindow_Instance_CurrentAssemblyListChanged(object sender, NotifyCollectionChangedEventArgs e)
+		{
+			if (e.OldItems == null)
+			{
+				return;
+			}
+			foreach (var tab in TabPages.ToArray())
+			{
+				var state = tab.GetState();
+				if (state == null || state.DecompiledNodes == null)
+				{
+					continue;
+				}
+				bool found = false;
+				foreach (var node in state.DecompiledNodes)
+				{
+					var assemblyNode = node.Ancestors().OfType<TreeNodes.AssemblyTreeNode>().FirstOrDefault();
```

Comment:
This will find the innermost `AssemblyTreeNode`, which may be a bundle member.
I think `LastOrDefault` would be more appropriate as it would handle the case where a bundle is removed from the assembly list.